File: python_backend/utils/types/chat_dto.py
from pydantic import BaseModel

class Message(BaseModel):
    role: str
    text: str

class ClientInfos(BaseModel):
    name: str
    user_id: int
    total_debit: float
    old_messages: list[Message] | None

class Proposal(BaseModel):
    message: Message
    entry: float = 0
    installments: int = 0
    accepted: bool = False

class Answer(BaseModel):
    require_input: bool = False
    is_finished: bool = False
    message: Message | None
    confirm_text: str = ""
    deny_text: str = ""

class ProposalResponse(BaseModel):
    proposal: Proposal | None = None
    answer: Answer

class Messages(BaseModel):
    messages: list[Proposal]
    answer: Answer

class ThreadInfos(BaseModel):
    client: ClientInfos

class UserProposal(BaseModel):
    user_id: int
    message: str
    value: float
    installments: int

File: nextjs_frontend/src/app/api/chat/chat.dto.ts
interface ApiMessage {
  role: string;
  text: string;
}

export interface ApiProposal {
  message: ApiMessage
  entry: number
  accepted: boolean
  installments: number
}

export interface ApiAnswer {
  require_input: boolean;
  confirm_text: string;
  deny_text: string;
  message: ApiMessage;
  is_finished: boolean;
}

export interface ApiProposalResponse {
  proposal: ApiProposal | null;
  answer: ApiAnswer;
}

export interface ApiPostResponse {
  messages: ApiProposal[];
  answer: ApiAnswer;
}

export const errorMsg = "Minha conexão está ruim... Poderia repetir?";

export const defaultApiProposalResponse: ApiProposalResponse = {
  proposal: null,
  answer: {
    require_input: false,
    is_finished: false,
    confirm_text: "",
    deny_text: "",
    message: {
      role: "assistant",
      text: errorMsg
    },
  }
}
